crates/cargo-gpu/src/metadata.rs

```diff
@@ -9,6 +10,28 @@ use serde_json::Value;
 pub struct Metadata;
 
 impl Metadata {
+    /// Resolve a package name to a crate directory.
+    pub fn resolve_package_to_shader_crate(package: &str) -> anyhow::Result<std::path::PathBuf> {
+        log::debug!("resolving package '{package}' to shader crate");
+        let metadata = MetadataCommand::new().exec()?;
```

Comment:
`MetadataCommand::new().exec()` runs the `cargo metadata` and parses it's json output *every time it is called*, it is not cached by `cargo_metadata` itself. We should be able to reuse the `Metadata` to resolve the `spirv-std` version as well.